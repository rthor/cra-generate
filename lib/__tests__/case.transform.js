'use strict'

const caseTransform = require('../case-transform')

describe('caseTransform', () => {
  test('returns empty string', () => {
    const emptyString = ''
    const actual = caseTransform(emptyString)
    expect(actual).toBe(emptyString)
  })

  test('throws on undefined transformation function', () => {
    const error = 'Invalid transform, allowed transform functions are:\n' +
      '[camelCase, constantCase, headerCase, paramCase, pascalCase, snakeCase]'
    expect(() => caseTransform('', 'foobar')).toThrowError(error)
  })

  describe('transform functions', () => {
    test('camelCase', () => {
      expect(caseTransform('test this', 'camelCase')).toBe('testThis')
    })

    test('constantCase', () => {
      expect(caseTransform('test this', 'constantCase')).toBe('TEST_THIS')
    })

    test('headerCase', () => {
      expect(caseTransform('test this', 'headerCase')).toBe('Test-This')
    })

    test('paramCase', () => {
      expect(caseTransform('test this', 'paramCase')).toBe('test-this')
    })

    test('pascalCase', () => {
      expect(caseTransform('test this', 'pascalCase')).toBe('TestThis')
    })

    test('snakeCase', () => {
      expect(caseTransform('test this', 'snakeCase')).toBe('test_this')
    })

    test('defaults to pascalCase', () => {
      const actual = caseTransform('test this')
      const expected = 'TestThis'
      expect(actual).toBe(expected)
    })
  })
})
