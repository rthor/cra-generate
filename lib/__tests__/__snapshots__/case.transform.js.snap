// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`caseTransform should default to pascal case 1`] = `"TestThis"`;

exports[`caseTransform should throw on undefined transformation function 1`] = `
"Invalid transform, allowed transform functions are:
[camelCase, constantCase, headerCase, paramCase, pascalCase, snakeCase]"
`;

exports[`caseTransform should use the correct transform function 1`] = `"testThis"`;

exports[`caseTransform should use the correct transform function 2`] = `"TEST_THIS"`;

exports[`caseTransform should use the correct transform function 3`] = `"Test-This"`;

exports[`caseTransform should use the correct transform function 4`] = `"test-this"`;

exports[`caseTransform should use the correct transform function 5`] = `"TestThis"`;

exports[`caseTransform should use the correct transform function 6`] = `"test_this"`;
